### PR TITLE
new cloud.GetUserOrProjectSecret for v2 secrets

### PR DESCRIPTION
**Pull request by alexcb**
Creates a new function to fetch v2 user or project secrets

Signed-off-by: Alex Couture-Beil <alex@earthly.dev>

### CODE REVIEW

File: cloud/secret_new.go [hunk 1]
```diff
@@ -86,6 +85,17 @@ func (c *client) GetUserSecret(ctx context.Context, secretName string) (*Secret,
 	return c.getSecretV2(ctx, fmt.Sprintf("/user/%s", secretName))
 }
 
+func (c *client) GetUserOrProjectSecret(ctx context.Context, path string) (*Secret, error) {
```
Comment (idodod) [inline cloud/secret_new.go:88]: Mind adding a comment?
Would `GetSecret` be a fitting name too?

File: cmd/earthly/secret_cmds.go [hunk 2]
```diff
@@ -168,16 +169,15 @@ func (app *earthlyApp) actionSecretsGetV2(cliCtx *cli.Context) error {
 		return err
 	}
 
-	secrets, err := cloudClient.ListSecrets(cliCtx.Context, path)
+	secret, err := cloudClient.GetUserOrProjectSecret(cliCtx.Context, path)
 	if err != nil {
+		if errors.Is(err, cloud.ErrNotFound) {
+			return errors.New("no secret found for that path")
```
Comment (idodod) [inline cmd/earthly/secret_cmds.go:175]: should we mention the path that was used in the error?

**Review by idodod — APPROVED** [review]
LGTM
made some minor comments.

File: cloud/secret_new.go (hunk 1 above)
Comment (alexcb) [reply cloud/secret_new.go:92]: > Would GetSecret be a fitting name too?

I want to make it clear that this is for the new project-based secrets and not the original secrets implementation.

Maybe when we fully remove support for the old secrets, we could then simplify these names.

File: cmd/earthly/secret_cmds.go (hunk 2 above)
Comment (alexcb) [reply cmd/earthly/secret_cmds.go:175]: seems a bit overkill, malformed paths should generally be caught higher up, e.g. parts will append a `/` prefix when none are given by the user.

**Review by nelsam — APPROVED** [review]
lgtm

File: cloud/secret_new.go [hunk 3]
```diff
@@ -66,6 +65,7 @@ func (c *client) ListSecrets(ctx context.Context, path string) ([]*Secret, error
 	return secrets, nil
 }
 
+// GetProjectSecret gets a secret from a project secret store
```
[the remark this answers is not in the document]
Comment (alexcb) [reply cloud/secret_new.go:68]: Good idea; however, I need to stay focused on the ecr-login work at the moment. Maybe we can do some client cleanup down the road after v0.7 is out.

**Pull request merged**